flow/src/main/java/com/sequenceiq/flow/core/AbstractAction.java

```diff
@@ -102,6 +113,23 @@ public void execute(StateContext<S, E> context) {
         });
     }
 
+    private void doExecute(StateContext<S, E> context, P payload, C flowContext, Map<Object, Object> variables, String flowStateName) throws Exception {
```

Comment:
this `doExecute` is a bit strange, doesn't have any extra info. Could you come up with a better naming, also we already have a `doExecute` call inside this method

Reply:
shit, there is one indeed. no extra info is needed, it does the execution. any idea what should I name it?

Reply:
renamed to executeAction